WordPress/Classes/ViewRelated/Reader/ReaderStreamViewController.swift

```diff
@@ -1882,6 +1874,66 @@ extension ReaderStreamViewController: ReaderPostUndoCellDelegate {
 }
 
 
+// MARK: - Select Interests Display
+private extension ReaderStreamViewController {
+    func displaySelectInterestsIfNeeded(_ content: ReaderContent) {
+        guard
+            FeatureFlag.readerImprovementsPhase2.enabled,
+            content.topicType == .discover
+        else {
+            // Hide the view if we're not on the discover tab, and it exists
+            self.selectInterestsViewController?.remove()
+            return
+        }
+
+        if self.selectInterestsViewController != nil {
```

Comment:
We're guarding in showSelectInterestsViewIfNeeded so why do we need to check if it's no nil here?